packages/services/api/src/modules/audit-logs/providers/audit-logs-manager.ts

```diff
@@ -0,0 +1,173 @@
+import { Injectable, Injector, Scope } from 'graphql-modules';
+import { z } from 'zod';
+import * as Sentry from '@sentry/node';
+import { User } from '../../../shared/entities';
+import { ClickHouse, sql } from '../../operations/providers/clickhouse-client';
+import { SqlValue } from '../../operations/providers/sql';
+import { OrganizationManager } from '../../organization/providers/organization-manager';
+import { Logger } from '../../shared/providers/logger';
+import { AuditLogEvent, auditLogSchema } from './audit-logs-types';
+
+const auditLogEventTypes = auditLogSchema.options.map(option => option.shape.eventType.value);
+
+export const AUDIT_LOG_CLICKHOUSE_OBJECT = z.object({
+  id: z.string(),
+  event_time: z.string(),
+  user_id: z.string(),
+  user_email: z.string(),
+  organization_id: z.string(),
+  event_action: z.enum(auditLogEventTypes as [string, ...string[]]),
+  metadata:
+    z.string().transform(x => auditLogSchema.parse(JSON.parse(x))) &&
+    z.string().transform(x => JSON.parse(x)),
+});
+
+export type AuditLogType = z.infer<typeof AUDIT_LOG_CLICKHOUSE_OBJECT>;
+
+const AuditLogClickhouseArrayModel = z.array(AUDIT_LOG_CLICKHOUSE_OBJECT);
+
+type AuditLogsArgs = {
+  selector: {
+    organization: string;
+  };
+  filter: {
+    userId: string;
+    startDate: Date;
+    endDate: Date;
+  };
+  pagination: {
+    limit: number;
+    offset: number;
+  };
+};
+
+type AuditLogRecordEvent = {
+  userId: string;
+  userEmail: string;
+  organizationId: string;
+  user: (User & { isAdmin: boolean }) | null;
+};
+
+@Injectable({
+  scope: Scope.Operation,
+  global: true,
+})
+export class AuditLogManager {
+  private logger: Logger;
+
+  constructor(
+    logger: Logger,
+    private clickHouse: ClickHouse,
+  ) {
+    this.logger = logger.child({ source: 'AuditLogManager' });
+  }
+
+  createLogAuditEvent(event: AuditLogEvent, record: AuditLogRecordEvent): void {
```

Comment:
@TuvalSimha please add a comment here regarding the usage of `void`, so other developers will understand that the audit log is implemented this way because we want to make sure it's fail-safe and doe not block the main flow